src/targets/gpu/lowering.cpp

```diff
@@ -325,7 +325,9 @@ struct miopen_apply
 
     static bool use_miopen_pooling(instruction_ref ins)
     {
-        if(enabled(MIGRAPHX_DISABLE_MIOPEN_POOLING{}))
+        std::set<shape::type_t> supported_types = {shape::float_type, shape::half_type};
+        if(enabled(MIGRAPHX_DISABLE_MIOPEN_POOLING{}) or
+           not contains(supported_types, ins->get_shape().type()))
```

Comment:
Just put the list inline instead of creating a set: `contains({shape::float_type, shape::half_type}, ins->get_shape().type())`